experimental/packages/opentelemetry-sdk-metrics-base/src/state/MeterProviderSharedState.ts

```diff
@@ -30,15 +30,21 @@ export class MeterProviderSharedState {
 
   metricCollectors: MetricCollector[] = [];
 
-  meterSharedStates: MeterSharedState[] = [];
+  meterSharedStates: Map<string, MeterSharedState> = new Map();
 
   constructor(public resource: Resource) {}
 
   getMeterSharedState(instrumentationLibrary: InstrumentationLibrary) {
-    // TODO: meter identity
-    // https://github.com/open-telemetry/opentelemetry-js/issues/2593
-    const meterSharedState = new MeterSharedState(this, instrumentationLibrary);
-    this.meterSharedStates.push(meterSharedState);
+    const id = this.instrumentationLibraryId(instrumentationLibrary);
+    let meterSharedState = this.meterSharedStates.get(id);
+    if (meterSharedState == null) {
+      meterSharedState = new MeterSharedState(this, instrumentationLibrary);
+      this.meterSharedStates.set(id, meterSharedState);
+    }
     return meterSharedState;
   }
+
+  instrumentationLibraryId(instrumentationLibrary: InstrumentationLibrary) {
```

Comment:
is this meant to be public?

Reply:
This struct is internal so I didn't put a visibility mark here. I think it can be moved in utils.ts.